gradoop-temporal/src/main/java/org/gradoop/temporal/model/impl/TemporalGraphCollection.java

```diff
@@ -201,38 +199,23 @@ public boolean isTransactionalLayout() {
   //----------------------------------------------------------------------------
 
   @Override
-  public GraphCollection toGraphCollection() {
-    final GraphCollectionFactory collectionFactory = this.config.getGraphCollectionFactory();
-    return collectionFactory.fromDataSets(
-      getGraphHeads().map(new TemporalGraphHeadToGraphHead(collectionFactory.getGraphHeadFactory())),
-      getVertices().map(new TemporalVertexToVertex(collectionFactory.getVertexFactory())),
-      getEdges().map(new TemporalEdgeToEdge(collectionFactory.getEdgeFactory())));
-  }
-
-  @Override
-  public TemporalGraphCollection callForCollection(
-    UnaryBaseGraphCollectionToBaseGraphCollectionOperator<TemporalGraphCollection> operator) {
+  public <T> T callForValue(UnaryBaseGraphCollectionToValueOperator<TemporalGraphCollection, T> operator) {
```

Comment:
Can the `callForValue` method be implemented in `BaseGraph` as a default method?

Reply:
`this` is needed but has different types in `BaseGraph` and `LogicalGraph`

Reply:
I see, you can ignore this then